Remotion/Core/UnitTests/Utilities/DateOnlyConverterTest.cs

```diff
@@ -0,0 +1,192 @@
+// SPDX-FileCopyrightText: (c) RUBICON IT GmbH, www.rubicon.eu
+// SPDX-License-Identifier: LGPL-2.1-or-later
+using System;
+using System.ComponentModel;
+using System.Globalization;
+using Moq;
+using NUnit.Framework;
+
+namespace Remotion.UnitTests.Utilities;
+
+using DateOnlyConverter = Remotion.Utilities.DateOnlyConverter;
+
+[TestFixture]
+public class DateOnlyConverterTest
+{
+  [Test]
+  [TestCase(typeof(DateTime))]
+  [TestCase(typeof(DateTime?))]
+  public void CanConvertFrom_DateTimeTypes_ReturnsTrue (Type testType)
+  {
+    var converter = new DateOnlyConverter();
+
+    var result = converter.CanConvertFrom(Mock.Of<ITypeDescriptorContext>(), testType);
+
+    Assert.That(result, Is.True);
+  }
+
+  [Test]
+  [TestCase(typeof(DateOnly))]
+  [TestCase(typeof(DateOnly?))]
+  [TestCase(typeof(string))]
+  [TestCase(typeof(int))]
+  [TestCase(typeof(DateOnlyConverter))]
+  public void CanConvertFrom_OtherTypes_ReturnsFalse (Type testType)
+  {
+    var converter = new DateOnlyConverter();
+
+    var result = converter.CanConvertFrom(Mock.Of<ITypeDescriptorContext>(), testType);
+
+    Assert.That(result, Is.False);
+  }
+
+  [Test]
+  [TestCase(typeof(DateTime))]
+  [TestCase(typeof(DateTime?))]
+  [TestCase(typeof(string))]
+  public void CanConvertTo_DateTimeTypes_ReturnsTrue (Type testType)
+  {
+    var converter = new DateOnlyConverter();
+
+    var result = converter.CanConvertTo(Mock.Of<ITypeDescriptorContext>(), testType);
+
+    Assert.That(result, Is.True);
+  }
+
+  [Test]
+  [TestCase(typeof(DateOnly))]
+  [TestCase(typeof(DateOnly?))]
+  [TestCase(typeof(int))]
+  [TestCase(typeof(DateOnlyConverter))]
+  public void CanConvertTo_OtherTypes_ReturnsFalse (Type testType)
+  {
+    var converter = new DateOnlyConverter();
+
+    var result = converter.CanConvertTo(Mock.Of<ITypeDescriptorContext>(), testType);
+
+    Assert.That(result, Is.False);
+  }
+
+  [Test]
+  public void ConvertFrom_NullableDateTime_WithValue ()
```

Comment:
We also need this test with not-nullable datetime. I know, in whitebox testing we could know that this will work but a refactoring might break this.

Reply:
https://github.com/re-motion/Framework/pull/1213#discussion_r1736101735 Thought we didnt need both of them? I don't understand